### PR TITLE
feat: init

### PR DESCRIPTION
**Pull request by owlot**
Set up the basis of our GCP environment.
Creates initial Terrafrom ServiceAccount, Identity Pool for GitHub Actions authentication, and Terraform state bucket.

### CODE REVIEW

**Review by GoodluckH — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -1,2 +1,76 @@
...
+If you don't have permission to look up the values for these variables, you will not have permissions to run local plans either.
+
+### Run Terraform plan
+```
+make tf-plan-services-global-all
+```
+
+## Initial setup
+
+In the beginning, there was no GCP infra...
```
Comment (amsmith-pro) [inline README.md:43]: :laughing:

File: src/modules/organization-iam/variables.tf [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    title       = string
+    description = string
+    permissions = list(string)
+  }))
+  default = {}
+}
+
+variable "org_role_bindings" {
+  description = "Map of pre-made GCP role bindings applied on the organization level"
+  type        = map(map(map(map(string))))
```
Comment (amsmith-pro) [inline src/modules/organization-iam/variables.tf]: :eyes:

Comment (amsmith-pro) [reply src/modules/organization-iam/variables.tf]: maybe have a comment explaining what each level of map is

File: src/modules/automation-setup/main.tf [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    "attribute.repository_owner_id" = "assertion.repository_owner_id"
+    "attribute.repository"          = "assertion.repository"
+  }
+  oidc {
+    issuer_uri = "https://token.actions.githubusercontent.com"
+  }
+}
+
+module "identity_provider_sa" {
+  source  = "owlot/service-account/google"
```
Comment (amsmith-pro) [inline src/modules/automation-setup/main.tf:60]: can this be a for_each locals.sa where locals.sa == ["owlot"] ? or should it only be linked to yours?

File: src/modules/vars/permissions.tf [hunk 4]
```diff
@@ -0,0 +1,543 @@
...
+      "logging.notificationRules.get",
+      "logging.notificationRules.list",
+      "logging.operations.get",
+      "logging.operations.list",
+      "logging.queries.create",
+      "logging.queries.delete",
+      "logging.queries.get",
+      "logging.queries.list",
+      "logging.queries.listShared",
+      "logging.queries.update",
```
Comment (amsmith-pro) [inline src/modules/vars/permissions.tf:331]: should this be allowed under _ro permissions?

File: src/modules/vars/permissions.tf [hunk 5]
```diff
@@ -0,0 +1,543 @@
...
+      "container.resourceQuotas.list",
+      "container.roleBindings.get",
+      "container.roleBindings.list",
+      "container.roles.get",
+      "container.roles.list",
+      "container.runtimeClasses.get",
+      "container.runtimeClasses.list",
+      "container.scheduledJobs.get",
+      "container.scheduledJobs.list",
+      "container.selfSubjectAccessReviews.create",
```
Comment (amsmith-pro) [inline src/modules/vars/permissions.tf:257]: confirming this is fine under _ro rules?

File: src/credentials.json
Comment (amsmith-pro) [inline src/credentials.json:1]: Confirming you want this empty credentials file committed to git as opposed to providing credentials through some other means

File: src/modules/automation-setup/main.tf (hunk 3 above)
Comment (owlot) [reply src/modules/automation-setup/main.tf:60]: This is the way to tell Terraform where to find the source of the module.
In this case I'm pointing it to a module I've published myself. We might fork it and host our own version under the Arrow org at some point, but for now I've just used mine.

https://registry.terraform.io/search/modules?q=owlot

File: src/modules/organization-iam/variables.tf (hunk 2 above)
Comment (owlot) [reply src/modules/organization-iam/variables.tf]: I removed this completely, we don't use it (yet) and this module will probably be moved outside of this repo together with the `iam` one. Only left the part that creates the custom roles now.

File: src/modules/vars/permissions.tf (hunk 4 above)
Comment (owlot) [reply src/modules/vars/permissions.tf:331]: I double checked, but yeah, in this case we do want to have the `update` permission as it allows the use to create log queries when browsing the logs in the GCP console.

File: src/modules/vars/permissions.tf (hunk 5 above)
Comment (owlot) [reply src/modules/vars/permissions.tf:257]: This was copied from the GCP provided `viewer` role which seems to have this permission, but I'm not 100% sure what it does. It's a GKE permission and it seems to be related to the k8s native authorization functionality.

File: src/credentials.json
Comment (owlot) [reply src/credentials.json:1]: I indeed did not intend to add this file, removed now.

**Review by amsmith-pro — DISMISSED** [review]

**Review by amsmith-pro — APPROVED** [review]

**Review by GoodluckH — APPROVED** [review]

**Pull request merged**